src/navigation/deep_cost_map_evaluator.cc

```diff
@@ -0,0 +1,392 @@
+//========================================================================
+//  This software is free: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License Version 3,
+//  as published by the Free Software Foundation.
+//
+//  This software is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  Version 3 in the file COPYING that came with this distribution.
+//  If not, see <http://www.gnu.org/licenses/>.
+//========================================================================
+/*!
+\file    deep_irl_evaluator.cc
+\brief   Path rollout evaluator using a NN learned via deep IRL.
+\author  Joydeep Biswas, Kavan Sikand, (C) 2021
+*/
+//========================================================================
+
+#include <math.h>
+#include <float.h>
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+#include <thread>
+
+#include "gflags/gflags.h"
+#include "math/line2d.h"
+#include "math/poses_2d.h"
+#include "eigen3/Eigen/Dense"
+#include "eigen3/Eigen/Geometry"
+#include "torch/torch.h"
+#include "torch/script.h"
+#include <chrono>
+
+#include "motion_primitives.h"
+#include "navigation_parameters.h"
+#include "constant_curvature_arcs.h"
+#include "ackermann_motion_primitives.h"
+#include "deep_cost_map_evaluator.h"
+#include "image_tiler.h"
+#include <opencv2/core/eigen.hpp>
+
+using std::min;
+using std::max;
+using std::string;
+using std::vector;
+using std::shared_ptr;
+using pose_2d::Pose2Df;
+using Eigen::Vector2f;
+using navigation::MotionLimits;
+using namespace geometry;
+using namespace math_util;
+using std::chrono::high_resolution_clock;
+using std::chrono::duration_cast;
+using std::chrono::duration;
+using std::chrono::milliseconds;
+
+const double COST_RECOMP_DT = 0.05;
```

Comment:
maybe could define this as some multiple of the actual `dt` provided as a config param? Basically just controls how often we recompute the cost map from our image observation in the parallel thread.